File: src/3280.convert-date-to-binary.3567/solution.ts
function convertDateToBinary (date: string): string {
    const year = +date.slice(0, 4);
    const month = +date.slice(5, 7);
    const day = +date.slice(8);
    return `${year.toString(2)}-${month.toString(2)}-${day.toString(2)}`;
}

File: src/3280.convert-date-to-binary.3567/solution.java
class Solution {
    public String convertDateToBinary(String date) {
        int year = Integer.parseInt(date.substring(0,4));
        int month = Integer.parseInt(date.substring(5,7));
        int day = Integer.parseInt(date.substring(8));

        return Integer.toBinaryString(year) + '-' + Integer.toBinaryString(month) + '-' + Integer.toBinaryString(day);

    }
}